### ISSUE TITLE
Ask how to choose what should be indexed in a TypeIndex

### ISSUE THREAD

**Issue by lecoqlibre** [Wimmics/solid-start#19]
TypeIndex should be used for "large piece of data". What does that mean?

**Comment by pchampin** [comment]
DISCLAIMER: this is my own understanding, based on some discussions I had with various people, but I never checked with them whether this particular interpretation is what they meant.

TypeIndex should be the entry points of client-to-client specifications. E.g:

* a chat application will look in the TypeIndex for instances of Chat ; then the application will use the C2C spec to find individual posts in each chat

* a photo sharing application will probably look in the TypeIndex for instances of  Album; then the application will use the C2C spec to find individual photos in each album (as well as comments, tags, etc.)

In other words, the TypeIndex is here to "bootstrap" other C2C specs.
